api/server/server.go

```diff
@@ -374,42 +374,31 @@ func getContainersTop(eng *engine.Engine, version version.Version, w http.Respon
 }
 
 func getContainersJSON(eng *engine.Engine, version version.Version, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	if err := parseForm(r); err != nil {
+	var err error
+	if err = parseForm(r); err != nil {
 		return err
 	}
-	var (
-		err  error
-		outs *engine.Table
-		job  = eng.Job("containers")
-	)
-
-	job.Setenv("all", r.Form.Get("all"))
-	job.Setenv("size", r.Form.Get("size"))
-	job.Setenv("since", r.Form.Get("since"))
-	job.Setenv("before", r.Form.Get("before"))
-	job.Setenv("limit", r.Form.Get("limit"))
-	job.Setenv("filters", r.Form.Get("filters"))
 
-	if version.GreaterThanOrEqualTo("1.5") {
-		streamJSON(job, w, false)
-	} else if outs, err = job.Stdout.AddTable(); err != nil {
-		return err
-	}
-	if err = job.Run(); err != nil {
-		return err
+	config := daemon.ContainersConfig{
+		All:     r.Form.Get("all") == "1",
+		Size:    r.Form.Get("size") == "1",
+		Since:   r.Form.Get("since"),
+		Before:  r.Form.Get("before"),
+		Filters: r.Form.Get("filters"),
 	}
-	if version.LessThan("1.5") { // Convert to legacy format
-		for _, out := range outs.Data {
-			ports := engine.NewTable("", 0)
-			ports.ReadListFrom([]byte(out.Get("Ports")))
-			out.Set("Ports", api.DisplayablePorts(ports))
-		}
-		w.Header().Set("Content-Type", "application/json")
-		if _, err = outs.WriteListTo(w); err != nil {
+
+	if tmp := r.Form.Get("limit"); tmp != "" {
+		if config.Limit, err = strconv.Atoi(tmp); err != nil {
```

Comment:
Are you ok with what’s in there now?
https://github.com/docker/docker/pull/12163/files#diff-44a527ea72ea9572e1cf71c7f1e2a0f0R390
-Doug

> On Apr 8, 2015, at 1:27 PM, Alexander Morozov <notifications@github.com> wrote:
> 
> In api/server/server.go <https://github.com/docker/docker/pull/12163#discussion_r27992179>:
> 
> >  	}
> > -	if version.LessThan("1.5") { // Convert to legacy format
> > -		for _, out := range outs.Data {
> > -			ports := engine.NewTable("", 0)
> > -			ports.ReadListFrom([]byte(out.Get("Ports")))
> > -			out.Set("Ports", api.DisplayablePorts(ports))
> > -		}
> > -		w.Header().Set("Content-Type", "application/json")
> > -		if _, err = outs.WriteListTo(w); err != nil {
> > +
> > +	if tmp := r.Form.Get("limit"); tmp != "" {
> > +		if config.Limit, err = strconv.Atoi(tmp); err != nil {
> ah, no, I meant to not use err variable from outer scope and not assign value to config.Limit directly, because you know :=/= confusions is pretty common. And here you don't need outer scope and I think that local scope easier to read. So, my snippet meant to be inside if != "".
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/12163/files#r27992179>.
>